PROTOCOL.md

```diff
@@ -100,6 +103,19 @@ By default, the reference implementation stores data files in directories that a
 This directory format is only used to follow existing conventions and is not required by the protocol.
 Actual partition values for a file must be read from the transaction log.
 
+### Change Data Files
+Change data files are stored in a directory at the root of the table named `_change_data`, and represent the changes for the table version they are in. For data with partition values, change data files are stored within the `_change_data` directory in their respective partitions (i.e. `_change_data/part1=value1/...`). Writers can _optionally_ produce these change data files as a consequence of operations that change underlying data, like `UPDATE`, `DELETE`, and `MERGE` operations to a Delta Lake table. Operations that only add new data should not produce separate change files. When available, change data readers should use the change data files instead of computing changes from the underlying data files.
```

Comment:
```suggestion
Change data files are stored in a directory at the root of the table named `_change_data`, and represent the changes for the table version they are in. For data with partition values, change data files are recommended to store within the `_change_data` directory in their respective partitions (i.e. `_change_data/part1=value1/...`). Writers can _optionally_ produce these change data files as a consequence of operations that change underlying data, like `UPDATE`, `DELETE`, and `MERGE` operations to a Delta Lake table. Operations that only add new data should not produce separate change files. When available, change data readers should use the change data files instead of computing changes from the underlying data files.
```

Reply:
this is a `should` and I think it's valid, no? if you are only adding new data ... then you should only add `AddFile`s and should not add `AddCDCFile`s. when we go to read ... we will see there are no `AddCDCFile`s in this version ... and so we will treat all the `AddFile`s as inserts.

so, to me, this "should" suggestion makes sense.

what do you disagree with @zsxwing ?

Reply:
It is okay to add a `AddCDCFile` for inserts as long as `_change_type` is set to `insert`?

Reply:
It is okay. That's why this is `should` instead of `must`.

Reply:
How about this?
> If an operation only add new data or remove existing data, a writer can write only data files and commit them in `AddFile`s or `RemoveFile`s without duplicating the data into change data files.

Reply:
LGTM